### DISCUSSION TITLE
euro format

### DISCUSSION THREAD

**Discussion by HFolkertH** [cincheo/daquota#22] [Q&A] [answered]
hello

What's the best way to show a number input als € 1.234,56?  
I tried toLocaleString() but that did not work. Perhaps because my experience with Javascriot is only beginner.
Or  should Dlite have a subtype Amount for Input components?

**Comment by renaudpawlak** [comment]
Clearly, that is still something missing in the current version. I have created an issue and will work on it ASAP.
A simple way is indeed to have a "currency" input kind. I think I will go this way for simplicity.

Maybe also a good opportunity to improve inputs (there are some features still missing):
- formatters
- input masks
- prepends and appends.

**Comment by HFolkertH** [comment]
would be great!

**Comment by renaudpawlak** [comment]
It turned out to be more complicated than expected but I have pushed version 0.9.11, which allows an input type "formatted-number". In the "Formatting" tab,  you can select the 'currency' type to format your number with the appropriate currency. It is not completely finished so expect some bumps. It will improve it ASAP.

**Comment by renaudpawlak** [comment, marked as answer]
Version 0.9.12 now supports currencies, units, percentages, and decimal formatting with the 'formatted-number' input type. It uses the Number.toLocaleString (Intl) API under the hood, so you will find all the possible options there). I will close the associated issue, however I guess we can go further by adding input masks.

**Reply by HFolkertH** [reply]
Thanks!!
